mesonbuild/backend/ninjabackend.py

```diff
@@ -2374,7 +2374,7 @@ def generate_link(self, target, outfile, outname, obj_list, linker, extra_args=[
         # options passed on the command-line, in default_options, etc.
         # These have the lowest priority.
         if not isinstance(target, build.StaticLibrary):
-            commands += compilers.get_base_link_args(self.environment.coredata.base_options,
+            commands += compilers.get_base_link_args(self.get_base_options_for_target(target),
```

Comment:
I'm not sure if this is right, but without this the buildtype option is not accessible in ``get_base_link_args``.

Reply:
This is correct, I have a patch sitting locally that fixes this in a bunch of other places too, and I will open a PR for it later. It's blocked on my other PRs.